common/lib/dependabot/dependency.rb

```diff
@@ -110,6 +110,67 @@ def display_name
       display_name_builder.call(name)
     end
 
+    def humanized_previous_version
+      # If we don't have a previous version, we *may* still be able to figure
+      # one out if a ref was provided and has been changed (in which case the
+      # previous ref was essentially the version).
+      if previous_version.nil?
+        return ref_changed? ? previous_ref : nil
+      end
+
+      if previous_version.match?(/^[0-9a-f]{40}$/)
+        return previous_ref if ref_changed? && previous_ref
```

Comment:
This line checks whether there's a reference available instead in the raw commit sha and use that if available (and has been changed).